glue/utils/array.py

```diff
@@ -175,3 +176,78 @@ def broadcast_to(array, shape):
     except AttributeError:
         array = np.asarray(array)
         return np.broadcast_arrays(array, np.ones(shape, array.dtype))[0]
+
+
+def find_chunk_shape(shape, n_max=None):
+    """
+    Given the shape of an n-dimensional array, and the maximum number of
+    elements in a chunk, return the largest chunk shape to use for iteration.
+
+    This currently assumes the optimal chunk shape to return is for C-contiguous
+    arrays.
+    """
+
+    if n_max is None:
+        return tuple(shape)
+
+    block_shape = []
+
+    max_repeat_remaining = n_max
+
+    for size in shape[::-1]:
+
+        if max_repeat_remaining > size:
+            block_shape.append(size)
+            max_repeat_remaining = max_repeat_remaining // size
+        else:
+            block_shape.append(max_repeat_remaining)
+            max_repeat_remaining = 1
+
+    return tuple(block_shape[::-1])
+
+
+def iterate_chunks(shape, chunk_shape=None, n_max=None):
+    """
+    Given a data shape and a chunk shape (or maximum chunk size), iteratively
+    return slice objects that can be used to slice the array.
+    """
+
+    if chunk_shape is None and n_max is None:
+        raise ValueError('Either chunk_shape or n_max should be specified')
+    elif chunk_shape is not None and n_max is not None:
+        raise ValueError('Either chunk_shape or n_max should be specified (not both)')
+    elif chunk_shape is None:
```

Comment:
`if chunk_shape is None !=  n_max is None:` if you want to be short

Reply:
I think I'll leave it as-is for clarity, I'd rather be more verbose in this case.